contribs/freight/src/main/java/org/matsim/freight/carriers/analysis/CarriersAnalysis.java

```diff
@@ -143,30 +162,44 @@ private void createScenarioForFreightAnalysis(String vehiclesPath, String networ
 		scenario = ScenarioUtils.loadScenario(config);
 
 		//load carriers according to freight config
-		CarriersUtils.loadCarriersAccordingToFreightConfig( scenario );
+		CarriersUtils.loadCarriersAccordingToFreightConfig(scenario);
 		this.carriers = CarriersUtils.addOrGetCarriers(scenario);
 	}
 
-	public void runCarriersAnalysis() throws IOException {
+	public void runDemandOfCarriersAnalysis() {
+		File folder = new File(String.valueOf(ANALYSIS_OUTPUT_PATH));
+		if(!folder.exists())
+			folder.mkdirs();
+		CarrierPlanAnalysis carrierPlanAnalysis = new CarrierPlanAnalysis(delimiter, carriers);
+		carrierPlanAnalysis.runAnalysisAndWriteStats(ANALYSIS_OUTPUT_PATH, CarrierPlanAnalysis.CarrierAnalysisType.onlyDemand);
+	}
+
+	/**
+	 * Run the analysis of the carriers based on the carrier file.
+	 *
+	 */
+	public void runBasicCarriersAnalysis() {
 
 		//Where to store the analysis output?
 		File folder = new File(String.valueOf(ANALYSIS_OUTPUT_PATH));
-		folder.mkdirs();
-		if (allCarriersHavePlans(carriers)) {
+		if(!folder.exists())
+			folder.mkdirs();
+		if (CarriersUtils.allCarriersWithDemandHavePlans(carriers)) {
 			CarrierPlanAnalysis carrierPlanAnalysis = new CarrierPlanAnalysis(delimiter, carriers);
-			carrierPlanAnalysis.runAnalysisAndWriteStats(ANALYSIS_OUTPUT_PATH);
-		}
-		else  {
+			carrierPlanAnalysis.runAnalysisAndWriteStats(ANALYSIS_OUTPUT_PATH, CarrierPlanAnalysis.CarrierAnalysisType.completeAnalysis);
+		} else {
 			log.warn("########## Not all carriers have plans. Skipping CarrierPlanAnalysis."); //TODO perhaps skipp complete analysis
 		}
 	}
-	public void runCompleteAnalysis() throws IOException {
-		runCarriersAnalysis();
+
+	public void runCompleteAnalysis() {
```

Comment:
I would change it a bit around and have one public `run((Carrier))Analysis(AnalysisType analysisType` method that internally switches depending on the enum AnalysisType instead of creating one public method for each type of analysis.

((If you like, you can also add methods with an empty signature, calling the new one with a default setting...))

And imo it should be checked here, that all necessary file are present for the selectedd type, e.g. the onlyCarrierAnalysis does not need the eventsfile, while the complete one needs it.

Reply:
implemented your suggestions